controller/controller.go

```diff
@@ -239,7 +250,10 @@ func (c *Controller) delete(obj Object) {
 }
 
 func (c *Controller) propagate(resourceEvents []ResourceEvent) {
-	topology := c.topology.Build(c.cache.List())
+	topology, err := c.topology.Build(c.cache.List())
+	if !c.topology.allowTopologyLoops && err != nil {
+		panic(err)
+	}
```

Comment:
`allowTopologyLoops` is now an implementation detail of the topology builder. Maybe let the `Build` func panic? Then, there would no need to return and catch errors here.

Reply:
Currently it is as high in the stack without having to deal with go routines. If it is to be moved I would move it far lower to https://github.com/Kuadrant/policy-machinery/blob/82898288a4c10f8e8bb4469248fcbd843601b988/machinery/topology.go#L123C1-L125C3 which is were the error is first created. Do you want to move it down to there?

Reply:
Frankly, now I'm leaning more towards not panicking at all: https://github.com/Kuadrant/policy-machinery/pull/21#discussion_r1747623819

But if we end up deciding to panic, then I would keep that call inside the `controller` package.